lib/authorize.js

```diff
@@ -1,26 +1,124 @@
 /*!
  * Copyright (c) 2020 Digital Bazaar, Inc. All rights reserved.
  */
-import * as authorization from 'auth-header';
-import {httpClient} from '@digitalbazaar/http-client';
-import noopLogger from './noopLogger.js';
+import {SECURITY_CONTEXT_V2_URL} from 'jsonld-signatures';
+import {parseSignatureHeader} from 'http-signature-header';
+import {suites} from 'jsonld-signatures';
+import {verifyCapabilityInvocation} from 'http-signature-zcap-verify';
+const {Ed25519Signature2018} = suites;
 
 /**
  * Authorizes an incoming request.
  *
  * @param {object} options - Options hashmap.
- * @param {object} [options.logger] - Optional logger.
- * @param {object} options.httpsAgent - httpsAgent.
+ * @param {object} options.documentLoader - Document loader used to load
+ *   DID Documents, capability documents, and JSON-LD Contexts.
+ * @param {string} [options.expectedAction] - The expected action for the
+ *   invoked capability.
+ * @param {string} options.expectedHost - The expected host for the invoked
+ *   capability.
+ * @param {string} [options.expectedAction] - The expected target for the
+ *   invoked capability.
+ * @param {Promise<string>} [options.getController] - Gets the controller URL
+ *   for the invoked target. If this value isn't specified, getRootCapability
+ *   must be specified.
+ * @param {Promise} [options.getRootCapability] - The expected target for the
+ *   invoked capability. If this value isn't specified, getController must be
+ *   specified.
+ * @param {class} [options.suite] - The expected cryptography suite to use when
+ *   verifying digital signatures.
  *
  * @returns {Function} Returns an Express.js middleware route handler.
  */
-export function authorize({
-  logger = noopLogger, httpsAgent
-}) {
-  _assertOk(httpsAgent, 'httpsAgent');
+export function authorizeZcapInvocation({
+  documentLoader, expectedAction, expectedHost, expectedTarget,
+  getCapabilityController, getRootCapability,
+  suite = new Ed25519Signature2018()
```

Comment:
@dlongley @dmitrizagidulin Is defaulting this to `Ed25519Signature2018` what is wanted here?  Also, is it true that there are no required parameters to the constructor here?

I'm also looking to get this switched over to this code:
- https://github.com/digitalbazaar/ed25519-signature-2018/blob/initial/lib/Ed25519Signature2018.js
- https://github.com/digitalbazaar/jws-linked-data-signature/blob/initial/lib/JwsLinkedDataSignature.js